libpromises/cf3parse.y

```diff
@@ -573,6 +574,88 @@ constraint:            constraint_id                        /* BUNDLE ONLY */
                                        // Intentional fall
                                    case SYNTAX_STATUS_NORMAL:
                                        {
+                                           const char *item = P.rval.item;
+                                           // convert @(x) to mergedata(x)
+                                           if (P.rval.type == RVAL_TYPE_SCALAR &&
+                                               (strcmp(P.lval, "data") == 0 || strcmp(P.lval, "template_data") == 0) &&
+                                               strlen(item) > 3 &&
+                                               item[0] == '@' &&
+                                               (item[1] == '(' || item[1] == '{'))
```

Comment:
I think that's OK.  In that case we simply try to parse with a function instead of directly, resulting in the same effect if those were not variable references.  I really don't want to write a unexpanded var ref detector--if you can suggest one, I'll use it...